cscs-checks/prgenv/affinity_check.py

```diff
@@ -3,15 +3,29 @@
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
+
 import os
+import json
+
 import reframe as rfm
 import reframe.utility.sanity as sn
+import reframe.utility.osext as osext
+from reframe.core.exceptions import SanityError
 
 
 class AffinityTestBase(rfm.RegressionTest):
-    def __init__(self, variant):
+    '''Base class for the affinity checks.
+
+    It reads a reference file for each valid system, which allows this base
+    class to figure out the processor's topology. The content of this reference
+    file is simply the output of the `lscpu -e` command. For more info on this,
+    see the `read_proc_topo` hook.
+    '''
+
+    def __init__(self):
         self.valid_systems = ['daint:gpu', 'daint:mc',
-                              'dom:gpu', 'dom:mc']
+                              'dom:gpu', 'dom:mc', 'eiger:mc',
+                              'ault:a64fx', 'ault:amdv100']
         self.valid_prog_environs = ['PrgEnv-gnu']
```

Comment:
do you expect the same result with other compilers ?

Reply:
> unless some compilers decide to not comply to the OMP standard

This is a regression, isn't it? In fact, we have seen differences in the past. Keep in mind that the compiler wrappers may also do some magic that's not visible immediately.

Reply:
I would probably put explicitly the different programming environments that we have (cray, gnu, intel, pgi, aocc)

Reply:
Because, for example, it's not so relevant for the `builtin`.

Reply:
Makes sense. I'll change that. Would it make sense to extend the `valid_prog_environs` syntax to allow something like 
`self.valid_prog_environs=['^builtin']`, which would use all the programming environments available except for `builtin`?

Reply:
`PrgEnv-aocc` has been left out for now (JG has already opened an internal issue on it).